FA2sp/Miscs/Hooks.SaveMap.cpp

```diff
@@ -19,10 +19,22 @@ DEFINE_HOOK(428D97, CFinalSunDlg_SaveMap, 7)
         GET(INIClass*, pINI, EAX);
         GET_STACK(CFinalSunDlg*, pThis, STACK_OFFS(0x3F4, 0x36C));
         REF_STACK(ppmfc::CString, filepath, STACK_OFFS(0x3F4, -0x4));
+        bool hasVersion = false;
 
         pThis->MyViewFrame.StatusBar.SetWindowText("Saving...");
         pThis->MyViewFrame.StatusBar.UpdateWindow();
 
+        int version = pINI->GetInteger("Basic", "Version");
+        if (version >= 1)
+            hasVersion = true;
+        else
+            version = 1;
+
+        if (ExtConfigs::SaveMap_OnlySaveMAP) {
```

Comment:
Should be better if you have some check here, for example, does it really have an extension right now? Can you make sure it just lengths 3? I think it's quite necessary.